src/main/java/org/apache/sling/rewriter/impl/ProcessorManagerImpl.java

```diff
@@ -327,61 +326,14 @@ private synchronized void updateProcessor(final String path) {
                 }
                 if ( index != -1 ) {
                     // we are already in the array
-                    if ( index == 0 ) {
-                        // we are the first, so remove the old, and add the new
                         this.orderedProcessors.remove(configs[index].config);
                         configs[index] = new ConfigEntry(path, config);
                         if ( config.isActive() ) {
                             this.orderedProcessors.add(config);
                             Collections.sort(this.orderedProcessors, new ProcessorConfiguratorComparator());
-                        }
-                    } else {
-                        // we are not the first, so we can simply exchange
-                        configs[index] = new ConfigEntry(path, config);
                     }
                 } else {
-                    // now we have to insert the new config at the correct place
-                    int insertIndex = 0;
-                    boolean found = false;
-                    while ( !found && insertIndex < configs.length) {
-                        final ConfigEntry current = configs[insertIndex];
-                        int currentIndex = -1;
-                        for(int i=0; i<searchPath.length; i++) {
-                            if ( current.path.startsWith(searchPath[i]) ) {
```

Comment:
If we don't find the new config inside ``configs``, we can't just pick the first config which starts with the search path ("/libs" and "/apps") and replace it with the new one...